mono/metadata/file-mmap-posix.c

```diff
@@ -358,7 +360,9 @@ open_memory_map (const char *c_mapName, int mode, gint64 *capacity, int access,
 		}
 
 		unlink (file_name);
+#ifdef HAVE_FTRUNCATE
 		unused = ftruncate (fd, (off_t)*capacity);
+#endif
```

Comment:
Else what?

Reply:
It doesn't seem like there would be any reason for there to be an else body. We silently discard the error there (the variable "unused"), and we don't seem to require that the file size truncation happen for correctness reasons. On the platforms without ftruncate we will simply not truncate.